lib/Storage/SecureViewWrapper.php

```diff
@@ -89,19 +89,25 @@ private function shouldSecure(string $path, ?IStorage $sourceStorage = null): bo
 			fclose($fp);
 		}
 
-		try {
-			$node = $this->rootFolder->get($this->mountPoint . $path);
-		} catch (NotFoundException) {
-			// If the file is just created we may need to check the parent as this is only just about figuring out if it is a share
-			$node = $this->rootFolder->get(dirname($this->mountPoint . $path));
+		$storage = $sourceStorage ?? $this;
+		$cacheEntry = $storage->getCache()->get($path);
+		if (!$cacheEntry) {
+			$parent = dirname($path);
+			if ($parent === '' || $parent === '.') {
+				throw new NotFoundException(sprintf('Could not find cache entry for path of %s within storage %s ', $path, $storage->getId()));
+			}
```

Comment:
I don't think is correct, I think that would then fail when trying to move a file to the root of a storage.

```php
if ($parent === '.') {
     $parent = '';
}
```

should be enough to handle the edge case

Reply:
adjusted to your suggestions anyways